addons/interaction/XEH_postInit.sqf

```diff
@@ -106,7 +106,7 @@ GVAR(isOpeningDoor) = false;
 {false},
 [20, [true, false, false]], false] call CBA_fnc_addKeybind;
 
-["isNotSwimming", {!underwater (_this select 0)}] call EFUNC(common,addCanInteractWithCondition);
+["isNotSwimming", {isTouchingGround (_this select 0)}] call EFUNC(common,addCanInteractWithCondition);
```

Comment:
`isTouchingGround  player` is false when in aircraft
this breaks the microDagr actions and button

maybe `(isTouchingGround _unit) || {(vehicle _unit) != _unit}` ?

Still broken during free-fall, but I don't think you could realistically do much during that.

Reply:
Hmh, I thought this does checks differently. Will look into it.

Reply:
Would checking height be sufficient?
```sqf
{isTouchingGround (_this select 0) || {(getPosASL (_this select 0)) select 2 > 0}}
```
Unsure if that would work with CUP ponds for example though, as they are above sea level, no clue if `surfaceIsWater` reports `true` for those either, will test tomorrow.

Reply:
Ooh just got an idea, animation state checking. Swimming animations start with one of those (Actions):
```sqf
[bdv","bsw","dve","sdv","ssw","swm"]
```

Reply:
surfaceIsWater reports true for inland water, I just checked on chernarus' black lake

Reply:
maybe if (animation->"actions" inheritsFrom "SwimmingActions")

Reply:
@alganthe ah, good to know! However I prefer the animation solution as animation is actually what prevents all those interactions for executing normally like they would on land. That way there is probably less inconsistency.

Reply:
@PabstMirror Doesn't seem like they inherit from that at all, maybe a select few?
``` sqf
[configFile >> "CfgMovesMaleSdr" >> "States" >> "abdvpercmstpsnonwrfldnon", true] call BIS_fnc_returnParents;
-> ["AbdvPercMstpSnonWrflDnon","AidlPercMstpSrasWrflDnon_G0S","AmovPercMstpSrasWrflDnon","AmovPercMstpSlowWrflDnon","StandBase","Default"]
```